compensations/src/test/java/org/jboss/narayana/compensations/functional/compensationScoped/CompensationScopedTest.java

```diff
@@ -147,6 +147,22 @@ public void testCompensationHandler() throws Exception {
         Assert.assertEquals(false, MyConfirmationHandler.dataAvailable);
     }
 
+    @Test
+    public void testTwoCompensationHandlers() throws Exception {
+
+        ParticipantCompletionCoordinatorRules.setParticipantCount(6);
+
+        MyCompensationHandler.expectedData = "blah";
+        MyConfirmationHandler.expectedData = "blah";
+
+        getBAControler().beginBusinessActivity();
+        service.doWork("blah");
+        service.doWork("blah");
+        getBAControler().cancelBusinessActivity();
+
+        Assert.assertEquals(true, MyCompensationHandler.dataAvailable);
```

Comment:
I was just about to mention that I can see where the pattern was copied from on line 147 ;)